hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java

```diff
@@ -755,6 +756,20 @@ public OzoneBucket getBucketDetails(
     verifyBucketName(bucketName);
     OmBucketInfo bucketInfo =
         ozoneManagerClient.getBucketInfo(volumeName, bucketName);
+
+    BucketLayout bucketLayout = bucketInfo.getBucketLayout();
```

Comment:
@JyotinderSingh IMHO listing `#linkBucket ` implementation can be moved out to a separate jira. Apart from the list operation the patch looks good to me and those changes are enough to unblock the long waiting patch HDDS-5959, which has to be pushed in asap to unblock all the follow up tasks Please create a new sub-task to implement the listing of linkBucket content and rebase this patch to include only the other changes. Does that sound good?

Reply:
Thanks for the suggestion @rakeshadr. I have removed the listBuckets modification for Link Buckets.